src/sync/synchronizer.rs

```diff
@@ -268,7 +268,7 @@ impl Synchronizer {
                     key: inner,
                     key_version: version,
                 };
-                let inner_map = self.in_memory_map.entry(outer_key).or_insert_with(HashMap::new);
+                let inner_map = self.in_memory_map.entry(outer_key).or_default();
```

Comment:
Updated this line as per the suggestion provided in the failed build log.
![image](https://github.com/pravega/pravega-client-rust/assets/122247323/7b06a35d-5d60-419b-9575-bd6498fdfe68)